external/kafka/src/test/scala/org/apache/spark/streaming/kafka/DirectKafkaStreamSuite.scala

```diff
@@ -381,3 +447,20 @@ object DirectKafkaStreamSuite {
     }
   }
 }
+
+private[streaming] class ConstantEstimator(rates: Double*) extends RateEstimator {
```

Comment:
@tdas I just committed a refactored version of the test that uses your `ConstantEstimator` and somewhat follows the test example in the other suite. I also did some other refactoring. The overall execution time is under a second now.  You mentioned some other remaining naming nits. I wasn't sure which remain, so let me know.